tests/select_related/tests.py

```diff
@@ -163,6 +163,18 @@ def test_chaining(self):
             self.assertEqual(obj.parent_1, parent_1)
             self.assertEqual(obj.parent_2, parent_2)
 
+    def test_select_related_after_values(self):
+        """
+        Running select_related() after calling values() raises a TypeError
+
+        On #25252 we add a friendly error message about why that happens.
+        """
+
+        message = "Cannot call select_related() after .values() or .values_list()"
+
+        with self.assertRaisesMessage(TypeError, message):
+            list(Species.objects.values('name').select_related('genus'))
```

Comment:
Lets also test that `values_list()` throws the same error. You can do it in the same test method.